From 627f981e148bb9f7dc01df97fa20fe6124f417f7 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Wed, 7 Nov 2018 17:43:10 +0000
Subject: [PATCH] overlays: uart0 - return GPIOs 14 and 15 to inputs

In the event that alternate pins are used (only useful on Compute
Modules), return the standard pins to inputs to avoid double-mapping
them.

See: https://www.raspberrypi.org/forums/viewtopic.php?p=1388713#p1316977

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 arch/arm/boot/dts/overlays/uart0-overlay.dts | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

--- a/arch/arm/boot/dts/overlays/uart0-overlay.dts
+++ b/arch/arm/boot/dts/overlays/uart0-overlay.dts
@@ -17,16 +17,17 @@
 		target = <&gpio>;
 		__overlay__ {
 			uart0_pins: uart0_pins {
-				brcm,pins = <14 15>;
-				brcm,function = <4>; /* alt0 */
-				brcm,pull = <0 2>;
+				brcm,pins = <14 15 14 15>;
+				brcm,function = <0 0 4 4>; /* alt0 */
+				brcm,pull = <0 0 0 2>;
 			};
 		};
 	};
 
 	__overrides__ {
-		txd0_pin = <&uart0_pins>,"brcm,pins:0";
-		rxd0_pin = <&uart0_pins>,"brcm,pins:4";
-		pin_func = <&uart0_pins>,"brcm,function:0";
+		txd0_pin = <&uart0_pins>,"brcm,pins:8";
+		rxd0_pin = <&uart0_pins>,"brcm,pins:12";
+		pin_func = <&uart0_pins>,"brcm,function:8",
+			   <&uart0_pins>,"brcm,function:12";
 	};
 };
